### PR TITLE
[fix][broker] Replicator failed to create remote producer

### PR DESCRIPTION
**Pull request by hanmz**
Fixes https://github.com/apache/pulsar/issues/21242

### Motivation

We will encounter an error if remote topic has a schema definition by `Schema.JSON(String.class)`  when start geo-replication.
For details: https://github.com/apache/pulsar/issues/21242

### Modifications

When creating the replicator producer, we use ` Schema.BYTES` instead of `Schema.AUTO_PRODUCE_BYTES()`.

There are two intentions here:
1. Use Schema.BYTES can avoid the above problems.
2. Whether schema is `Schema.BYTES` or `Schema.AUTO_PRODUCE_BYTES()`, it does not affect the replicate schemas across multiple.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by Jason918 — COMMENTED** [review]
Can you add a unit test to make sure this fix can solve the issue?

**Comment by hanmz** [comment]
> Can you add a unit test to make sure this fix can solve the issue?

Okay, I've added a unit test.

**Comment by hanmz** [comment]
> Can you add a unit test to make sure this fix can solve the issue?

Okay, I've added a unit test

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by Technoboy- — APPROVED** [review]

**Review by poorbarcode — CHANGES_REQUESTED** [review]
Since this change is important, I need time to review it. Submit a Request Change to avoid it merged before

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public AbstractReplicator(String localCluster, Topic localTopic, String remoteCl
                         localTopicName + "-->" + remoteTopicName,
                 StringUtils.equals(localCluster, remoteCluster) ? localCluster : localCluster + "-->" + remoteCluster
         );
-        this.producerBuilder = replicationClient.newProducer(Schema.AUTO_PRODUCE_BYTES()) //
+        this.producerBuilder = replicationClient.newProducer(Schema.BYTES) //
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java:92]: Seems it is an issue of Schema Service, @liangyepianzhou Could you take a look?

Screenshot 2023-10-25 at 10 44 44

```
2023-10-25T10:46:27,178 - WARN  - [pulsar-io-105-14:AbstractReplicator@151] - [persistent://pulsar/ns/testReplicatorProducerByBytesSchema-0a1e1c3c-c8f0-4fd6-9121-fac5f109abda | r2-->r1] Failed to create remote producer (org.apache.avro.AvroRuntimeException: Not a record: "string"), retrying in 0.198 s
java.util.concurrent.CompletionException: org.apache.avro.AvroRuntimeException: Not a record: "string"
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1159)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.apache.pulsar.client.impl.BinaryProtoLookupService.lambda$getSchema$11(BinaryProtoLookupService.java:282)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.apache.pulsar.client.impl.ClientCnx.handleGetSchemaResponse(ClientCnx.java:923)
	at org.apache.pulsar.common.protocol.PulsarDecoder.channelRead(PulsarDecoder.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.flush.FlushConsolidationHandler.channelRead(FlushConsolidationHandler.java:152)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1471)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1345)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1385)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.avro.AvroRuntimeException: Not a record: "string"
	at org.apache.avro.Schema.getFields(Schema.java:283)
	at org.apache.pulsar.client.impl.schema.generic.GenericSchemaImpl.<init>(GenericSchemaImpl.java:43)
	at org.apache.pulsar.client.impl.schema.generic.GenericJsonSchema.<init>(GenericJsonSchema.java:39)
	at org.apache.pulsar.client.impl.schema.generic.GenericSchemaImpl.of(GenericSchemaImpl.java:79)
	at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.getSchema(AutoConsumeSchema.java:260)
	at org.apache.pulsar.client.impl.PulsarClientImplementationBindingImpl.getSchema(PulsarClientImplementationBindingImpl.java:256)
	at org.apache.pulsar.client.api.Schema.getSchema(Schema.java:463)
	at org.apache.pulsar.client.impl.PulsarClientImpl.lambda$createProducerAsync$0(PulsarClientImpl.java:361)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	... 43 more
```

Comment (poorbarcode) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java:92]: The implementation of simple type for the JSON Schema was not supported, this is the root cause. Do not modify here

**Pull request closed without merging**